src/server/engine_shard_set.cc

```diff
@@ -80,20 +80,22 @@ EngineShard::Stats& EngineShard::Stats::operator+=(const EngineShard::Stats& o)
 // 3. in case the above is OK, make sure that we have a "gap" between usage and commited memory
 // (control by commit_use_threshold flag)
 bool EngineShard::DefragTaskState::IsRequired() const {
-  const uint64_t threshold_mem = max_memory_limit * GetFlag(FLAGS_mem_defrag_threshold);
+  const int64_t threshold_mem = max_memory_limit * GetFlag(FLAGS_mem_defrag_threshold);
   const double commit_use_threshold = GetFlag(FLAGS_commit_use_threshold);
 
   if (cursor > kCursorDoneState) {
     return true;
   }
 
-  uint64_t commited = GetMallocCurrentCommitted();
+  // can be negative due to weird accounting of mimalloc.
+  int64_t commited = GetMallocCurrentCommitted();
```

Comment:
I think that in case we have a negative number here, we can be off completely. maybe it would be better to abort in this case? If we cannot be sure about this, being positive number, then we would make wrong calculations. In any case the fix itself is good since GetMallocCurrentCommitted indeed returns a sign number.

Reply:
if comitted is negative we do abort via the condition below.